src/components/datagrid/data_grid.tsx

```diff
@@ -129,22 +130,20 @@ function renderPagination(props: EuiDataGridProps) {
 }
 
 export const EuiDataGrid: FunctionComponent<EuiDataGridProps> = props => {
+  const interactiveCellId = htmlIdGenerator()();
```

Comment:
value of `interactiveCellId` needs to be stored in the component's state, otherwise future render passes will generate and use a new random id

```suggestion
  const [interactiveCellId] = useState(htmlIdGenerator()());
```